core/build.gradle

```diff
@@ -54,44 +122,43 @@ dependencies {
     compile ([group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5.3'])
     compile ([group: 'com.google.guava', name: 'guava', version: '20.0'])
     compile ([group: 'com.google.code.gson', name: 'gson', version: '2.8.2'])
+    compile ([group: 'com.google.protobuf', name:'protobuf-java', version:'3.6.1'])
+    compile ([group: 'io.grpc', name: 'grpc-stub', version: '1.14.0'])
+    compile ([group: 'io.grpc', name: 'grpc-protobuf', version: '1.14.0'])
+    runtime ([group: 'io.grpc', name: 'grpc-netty-shaded', version: '1.14.0'])
     testCompile group: 'junit', name: 'junit', version: '4.12'
     testCompile group: 'org.mockito', name: 'mockito-all', version: '1.10.19'
     testCompile group: 'com.google.code.gson', name: 'gson', version: '2.8.2'
+    testCompile group: 'io.grpc', name:'grpc-testing', version:'1.14.0'
 }
 
 ext.dependenciesPrefix = 'com.microsoft.applicationinsights.core.dependencies'
 shadowJar {
     classifier=''
+    mergeServiceFiles()
     relocate 'org.apache.http', "${dependenciesPrefix}.http"
     relocate 'eu.infomas.annotation', "${dependenciesPrefix}.annotation"
     relocate 'org.apache.commons', "${dependenciesPrefix}.apachecommons"
-    relocate 'com.google.common', "${dependenciesPrefix}.googlecommon"
     relocate 'javax.annotation', "${dependenciesPrefix}.javaxannotation"
-    relocate 'com.google.gson', "${dependenciesPrefix}.gson"
-    relocate 'com.google.thirdparty', "${dependenciesPrefix}.google.thirdparty"
+    relocate 'com.google', "${dependenciesPrefix}.google"
+    relocate 'io.grpc', "${dependenciesPrefix}.io.grpc"
+    relocate 'io.opencensus', "${dependenciesPrefix}.io.opencensus"
```

Comment:
Do we have a dependency on OpenCensus?

Reply:
it's a dependency of grpc.

partial output of `gradlew :core:dependencies`
```
\--- io.grpc:grpc-protobuf:1.14.0
     +--- io.grpc:grpc-core:1.14.0 (*)
     +--- com.google.protobuf:protobuf-java:3.5.1 -> 3.6.1
     +--- com.google.guava:guava:20.0
     +--- com.google.api.grpc:proto-google-common-protos:1.0.0
     \--- io.grpc:grpc-protobuf-lite:1.14.0
          +--- io.grpc:grpc-core:1.14.0 (*)
          \--- com.google.guava:guava:20.0
```

Reply:
Wow. This is amazing. I didn't new gRPC(which is communication protocol) will have OC embedded in it. Doesn't make any sense. It feels like a overuse of power :)

Reply:
Maybe it logs metrics?

Reply:
It's open census api. Anyway, no alternative I think so keep it repackaged.